lms/djangoapps/certificates/tests/test_api.py

```diff
@@ -1221,6 +1227,12 @@ def test_available_vs_display_date(
             assert maybe_avail == available_date_for_certificate(self.course, self.certificate)
             assert self.certificate.modified_date == display_date_for_certificate(self.course, self.certificate)
 
+            # With a certificate date override, display date returns the override, available date ignores it
+            self.certificate.date_override = MockCertificateDateOverride()
```

Comment:
Apparently! The rest of the test uses a `MockGeneratedCertificate` instead of the `GeneratedCertificateFactory`. It seems these need to be consistent; meaning: using the `CertificateDateOverrideFactory` requires a "real" (or factory-ized) `GeneratedCertificate`. I didn’t feel burdened to change the whole test to use factories instead of these mocks.